src/Microsoft.Tye.Hosting/DockerRunner.cs

```diff
@@ -219,210 +219,215 @@ private async Task StartContainerAsync(Application application, Service service,
 
             async Task RunDockerContainer(IEnumerable<(int ExternalPort, int Port, int? ContainerPort, string? Protocol)> ports)
             {
-                var hasPorts = ports.Any();
+                while (!dockerInfo.StoppingTokenSource.IsCancellationRequested)
```

Comment:
Why is there a loop here?

Reply:
When I implement the monitor, it might decide to kill a container because it's unhealthy. In that case I'd want the `DockerRunner` to recreate the replica.
Another way to do it, is to kill the container by calling `docker restart`, and then there is no need for that loop. what do you think?

Reply:
The argument behind the former, is that it's consistent with how I'll do it in `ProcessRunner`

Reply:
The docker runner already restarts the replica albiet with the same name.

Reply:
@davidfowl I think that the problem with doing `docker restart` is that the health monitor won't be able to tell that the container was restarted (because there won't be `Stopped` and `Started` events)

Reply:
From what I can tell with this loop, you would call `docker rm` and other commands I don't think you want to run: https://github.com/dotnet/tye/blob/65c7ae0d80d0b3749127839732cf7d1ac9926b79/src/Microsoft.Tye.Hosting/DockerRunner.cs#L429

The restart loop for rerunning docker is here: https://github.com/dotnet/tye/blob/65c7ae0d80d0b3749127839732cf7d1ac9926b79/src/Microsoft.Tye.Hosting/DockerRunner.cs#L382. Can you modify this loop instead?

Reply:
@jkotalik yeah, makes sense. I'll try doing it that way.

Reply:
@jkotalik changed the `DockerRunner` to do `docker restart` when `StoppingTokenSource` is canceled.